byron/chain/executable-spec/byron-spec-chain.cabal

```diff
@@ -18,7 +18,7 @@ flag development
     default: False
     manual: True
 
-common base                         { build-depends: base                             >= 4.11       && < 4.15     }
```

Comment:
Byron still builds quite happily on older `base`, so this constraint seems wrong

Reply:
Node, wallet and other downstream projects have dropped support for `ghc-8.6.5`, so we don't need to support this version for that purpose.

What reasons do we have to support older `base`?

Reply:
I'm not proposing we support it, just that we don't arbitrarily say it won't build.

Reply:
There are philosophical difference at play, but also practical ones.

There are two approaches I could see us taking:

* We could tighten bounds when we discover something doesn't work
* We could tighten bounds to reflect what we're willing to commit to maintaining in CI

If we choose the first approach, without automated CI, it's unlikely anyway will do the checking so the bounds will end up never getting updated, which undermines the usefulness of bounds.

One of the motivations we have for dropping `ghc-8.6.5` in CI is that solving for both `ghc-8.6.5` and `ghc-8.10.x` results in the cabal solver struggling in downstream projects that have more dependencies.

In `cardano-node` sometimes the solver fails altogether and we try to coax the solver to working by introducing `maxbackjumps` and `reorder-goals` which adds 11 minutes to the solver time.  This is a cost we having been paying for builds both in CI and also locally in the developer environment.

I've been asked to tighten up the lower bounds in `cardano-prelude` and `cardano-base` as well, which means `cardano-ledger-specs` may not build with this compilers after updating these dependencies.

I can certainly update this PR not make changes to `base` lower bounds, but this might be futile unless we go and rollback those changes in upstream projects.

Reply:
I have push an additional commit to revert the lower bounds changes.  If we decide to keep them, I will squash appropriately.

Reply:
OK, fair enough, I guess we should be consisstent.